cli/types/index.d.ts

```diff
@@ -174,757 +182,739 @@ declare namespace Cypress {
+     *
      * @see https://on.cypress.io/last
      */
-    last(options?: LoggableTimeoutable): Chainable
+    last<E extends Node = HTMLElement>(options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<E>>
 
     /**
+     * Get the global `window.location` object of the page that is currently active.
+     *
      * @see https://on.cypress.io/location
      */
-    location(options?: Loggable): Chainable
+    location(options?: Partial<Loggable & Timeoutable>): Chainable<Location>
+    location(key: string, options?: Partial<Loggable & Timeoutable>): Chainable<Location>
 
     /**
+     * Print a message to the Cypress Command Log.
+     *
      * @see https://on.cypress.io/log
      */
-    log(message: string, args: any): Chainable
+    log(message: string, ...args: any[]): Chainable<null>
 
     /**
+     * Get the immediately following sibling of each DOM element within a set of DOM elements.
+     *
      * @see https://on.cypress.io/next
      */
-    next(options?: LoggableTimeoutable): Chainable
-    next(selector: string, options?: LoggableTimeoutable): Chainable
+    next<K extends keyof HTMLElementTagNameMap>(selector: K, options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<HTMLElementTagNameMap[K]>>
+    next<E extends Node = HTMLElement>(options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<E>>
+    next<E extends Node = HTMLElement>(selector: string, options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<E>>
 
     /**
+     * Get all following siblings of each DOM element in a set of matched DOM elements.
+     *
+     * @see https://on.cypress.io/nextall
+     */
+    nextAll<K extends keyof HTMLElementTagNameMap>(selector: K, options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<HTMLElementTagNameMap[K]>>
+    nextAll<E extends HTMLElement = HTMLElement>(options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<E>>
+    nextAll<E extends HTMLElement = HTMLElement>(selector: string, options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<E>>
+
+    /**
+     * Get all following siblings of each DOM element in a set of matched DOM elements up to, but not including, the element provided.
+     *
+     * @see https://on.cypress.io/nextuntil
+     */
+    nextUntil<K extends keyof HTMLElementTagNameMap>(selector: K, options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<HTMLElementTagNameMap[K]>>
+    nextUntil<E extends HTMLElement = HTMLElement>(options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<E>>
+    nextUntil<E extends HTMLElement = HTMLElement>(selector: string, options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<E>>
+
+    /**
+     * Filter DOM element(s) from a set of DOM elements. Opposite of `.filter()`
+     *
      * @see https://on.cypress.io/not
      */
-    not(selector: string, options?: LoggableTimeoutable): Chainable
+    not<E extends Node = HTMLElement>(selector: string, options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<E>>

```

Comment:
We can't narrow the type down much with a `.not`. I'm not even sure a generic should be used here. `HTMLElement` is what all elements would have in common. I think I'll remove the generic from this.